x/session/keeper/keeper.go

```diff
@@ -60,12 +60,11 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 func (k Keeper) BeginBlocker(ctx sdk.Context) {
 	// ctx.BlockHeader().LastBlockId.Hash is the hash of the last block committed
 	hash := ctx.BlockHeader().LastBlockId.Hash
-	// ctx.BlockHeader().Height is the height of the current block being processed,
-	// decrementing it by 1 gives us the height of the last block committed.
-	height := ctx.BlockHeader().Height - 1
-	// If height is 0, ctx.BlockHeader().LastBlockId.Hash will be nil,
+	// ctx.BlockHeader().Height is the height of the last committed block.
+	height := ctx.BlockHeader().Height
+	// If height is 1, ctx.BlockHeader().LastBlockId.Hash will be nil,
 	// fall back to an empty byte slice.
-	if height == 0 {
+	if height == 1 {
```

Comment:
Why is this 1?

If the last commit block is 1, it means we have committed a real block. Genesis is 0.

Reply:
Yes, block `1` should behave just like the others, but for some reason it doesn't have its `ctx.BlockHeader().LastBlockId` populated and the `Hash` has its 0 value that panics when the CosmosSDK store checks for valid values.

The empty byte slice seems to fix it (but we can assign it any valid byte slice value)

![image](https://github.com/pokt-network/poktroll/assets/231488/7ac5a02f-b493-4322-a48e-df20d2d97d05)

Will have a better comment if you are OK with the solution.

Reply:
I appreciate the workaround and the detailed comment, but I still don't understand "why". maybe it's a bug in cosmos or rollkit? I want to be convinced that we're not bandainding this but actually understand why its the solution.

For example, see this comment where I reached out to cosmos when I didn't understand something in the past: https://github.com/pokt-network/poktroll/pull/323#discussion_r1463541516

Reply:
![image](https://github.com/pokt-network/poktroll/assets/231488/eda39961-a3bc-4c3d-8460-562b117ff6ce)

This should explain why there's no hash for block `1`. Will update the comments accordingly.

We are also assuming that genesis starts at block `0`. We should revisit our session start/end calculations and their corresponding tests. Will create a ticket for that.